### PR TITLE
feat(pod): add fail pod add fail pod experiment

### PR DESCRIPTION
**Pull request by Ghostbaby**
### Describe what this PR does / why we need it
add fail pod add fail pod experiment

### Describe how you did it
所选的pod在指定的时间段内将不可用，CR删除之后，将销毁实验容器恢复之前镜像。

### CODE REVIEW

**Comment by xcaspar** [comment]
@Ghostbaby Thank you for you contribution, please add usage about the feature.

File: exec/pod/failexp.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	spec.BaseExpActionCommandSpec
+}
+
+func NewFailPodActionSpec(client *channel.Client) spec.ExpActionCommandSpec {
+	return &FailPodActionSpec{
+		spec.BaseExpActionCommandSpec{
+			ActionMatchers: []spec.ExpFlagSpec{},
+			ActionFlags: []spec.ExpFlagSpec{
+				&spec.ExpFlag{
+					Name:   "fail",
```
Comment (xcaspar) [inline exec/pod/failexp.go:42]: Why add the `fail` flag?

File: exec/pod/failexp.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	for i, container := range pod.Spec.Containers{
+		key := fmt.Sprintf("%s-%s","failPod", container.Name)
+		if pod.Annotations == nil {
+			pod.Annotations = make(map[string]string)
+		}
+		if isAnnotationExist(pod.Annotations, key) {
+			continue
+		}
+		pod.Annotations[key] = container.Image
+		pod.Spec.Containers[i].Image = "gcr.io/google-containers/pause:latest"
```
Comment (xcaspar) [inline exec/pod/failexp.go:183]: I think it is better to understand and identify whether the image can be changed the original image to `pod.Spec.Containers[i].Image-fault-injection`. For example  xxxx:1.0-fault-injection

File: exec/pod/failexp.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+}
+
+// failPod will exec failPod experiment
+func (d *FailPodActionExecutor) failPod(ctx context.Context,  pod *v1.Pod) error {
+	for i, container := range pod.Spec.Containers{
+		key := fmt.Sprintf("%s-%s","failPod", container.Name)
+		if pod.Annotations == nil {
+			pod.Annotations = make(map[string]string)
+		}
+		if isAnnotationExist(pod.Annotations, key) {
```
Comment (xcaspar) [inline exec/pod/failexp.go:179]: True means skipping to inject fault?

Comment (Ghostbaby) [reply exec/pod/failexp.go:179]: yes, if this annotation key exists, reconciler will skip injecting this pod.

File: exec/pod/failexp.go (hunk 1 above)
Comment (Ghostbaby) [reply exec/pod/failexp.go:42]: 😂    I will remove this flag.

**Pull request merged**